singer_sdk/plugin_base.py

```diff
@@ -44,7 +65,10 @@
 class PluginBase(metaclass=abc.ABCMeta):
     """Abstract base class for taps."""
 
-    name: str  # The executable name of the tap or target plugin.
+    name: str  # The executable name of the tap or target plugin. e.g. tap-foo
+    package_name: str | None = (
+        None  # The package name of the plugin. e.g meltanolabs-tap-foo
+    )
```

Comment:
As a hack day project, maybe we could implement something that determines the package name given the executable name. As far as I know there's no easy way to do this, but I think it's still possible.